backends/dpdk/dpdkAsmOpt.h

```diff
@@ -290,6 +290,58 @@ class ShortenTokenLength : public Transform {
     }
 };
 
+// This pass identify redundant copies/moves and eliminate it
+// and propogate original sources.
+class CopyPropagationAndElimination : public Transform {
+    std::unordered_map<cstring, int> copyInfo;
+    std::unordered_map<cstring, int> usesInfo;
+    std::unordered_map<cstring, int> defInfo;
+    std::unordered_map<cstring /*def*/, const IR::Expression* /*use*/> replacementMap;
+    std::set<cstring> dontEliminate;
+
+    bool haveSingleUseDefCopy(cstring str) {
+        return copyInfo[str] == 1 && defInfo[str] == 1 && usesInfo[str] == 1;
+    }
+
+    const IR::Expression* getIrreplaceableExpr(cstring str);
+    void markUseDef(const IR::DpdkBinaryStatement *b);
+    void markUseDef(const IR::DpdkMovStatement *mv);
+    void markUseDef(const IR::DpdkUnaryStatement *u);
+    void markUseDef(const IR::DpdkCastStatement *c);
+    void markUseDef(const IR::DpdkJmpCondStatement *b);
+    const IR::Expression* replaceIfCopy(const IR::Expression *expr);
+
+    IR::IndexedVector<IR::DpdkAsmStatement>
+    copyPropAndDeadCodeElim(IR::IndexedVector<IR::DpdkAsmStatement> stmts);
+    void clear() {
+        replacementMap.clear();
```

Comment:
clear all data structure as we consider every action and apply block a independent block.